tests/unit-tests/test-class-quiz.php

```diff
@@ -1248,6 +1248,35 @@ function testSaveUserAnswersQuestionsAskedNotOverwriteable() {
 
 	}
 
+	/**
+	 * Ensure that when the quiz pagination is enabled, that is reflected in the loop object.
+	 *
+	 * @covers Sensei_Quiz::start_quiz_questions_loop
+	 * @group  questions
+	 */
+	public function testQuizQuestionsLoopShouldPaginate() {
+		/* Arrange */
+		global $post, $sensei_question_loop;
+
+		$lesson_id = $this->factory->lesson->create();
+		$quiz_id   = $this->factory->maybe_create_quiz_for_lesson( $lesson_id );
+		$post      = get_post( $quiz_id );
+
+		$this->factory->question->create_many( 10, [ 'quiz_id' => $quiz_id ] );
 
+		update_post_meta(
+			$quiz_id,
+			'_pagination',
+			wp_json_encode( [ 'pagination_number' => 2 ] )
+		);
+
+		/* Act */
+		Sensei_Quiz::start_quiz_questions_loop();
+
+		/* Assert */
+		$this->assertEquals( 2, $sensei_question_loop['posts_per_page'] );
```

Comment:
Small suggestion, but could consider adding a third parameter to this calls so that if only 1 were to fail, it would be easier to identify which one that is.

Reply:
Nice tip. Updated in https://github.com/Automattic/sensei/commit/878aa8d8b84d348fca613fead9436a750ea5be40.